src/Pages/Nodes.razor

```diff
@@ -235,9 +284,23 @@
         }
         else
         {
-            arg.Item.UpdateDatetime = DateTimeOffset.Now;
-            arg.Item.CreationDatetime = DateTimeOffset.Now;
-            var addResult = await NodeRepository.AddAsync(arg.Item);
+            Node newNode = arg.Item as Node;
+            (bool, string?) addResult;
+            var restoredNode = await NodeRepository.GetByPubkey(arg.Item.PubKey);
+         
+            if (restoredNode != null)
+            {
+                newNode = restoredNode;
+                newNode.Endpoint = arg.Item.Endpoint;
+                newNode.PubKey = arg.Item.PubKey;
+                newNode.ChannelAdminMacaroon = arg.Item.ChannelAdminMacaroon;
+                newNode.IsNodeEnabled = true;
+                addResult =  NodeRepository.Update(newNode);
```

Comment:
what about name, description, returning funds wallet, autosweep?

Reply:
They were assigned from restoreNode.

Reply:
what if the person renamed the node, or activated autosweep when they re-added the node?

Reply:
Fixed as we discussed.